### PR TITLE
Enable or disable elasticsearch from values

### PR DESCRIPTION
**Pull request by ydbondt**

### CODE REVIEW

File: apim/3.x/templates/api/api-configmap.yaml [hunk 1]
```diff
@@ -116,6 +116,7 @@ data:
     analytics:
       type: elasticsearch
       elasticsearch:
+        enabled: {{ .Values.es.enabled }}
```
Comment (ydbondt) [inline apim/3.x/templates/api/api-configmap.yaml:119]: Apparently this can't be disabled in api config ?

Comment (brasseld) [reply apim/3.x/templates/api/api-configmap.yaml:119]: Hi @ydbondt 

For now, analytics can't be disable for mgmt-api.
You don't want analytics ?

**Review by brasseld — CHANGES_REQUESTED** [review]
Looking for feedback

File: apim/3.x/templates/api/api-configmap.yaml (hunk 1 above)
Comment (ydbondt) [reply apim/3.x/templates/api/api-configmap.yaml:119]: It is not that we don't want analytics, but when we do a proof of concept we don't want to be forced to install a rather big component like elasticsearch.

**Comment by arne-heil** [comment]
Please implement this feature. We could really use a minimal footprint with at least Elasticsearch as an optional choice. Thanks in advance :smile:

**Comment by brasseld** [comment]
> Please implement this feature. We could really use a minimal footprint with at least Elasticsearch as an optional choice. Thanks in advance 😄

Hi @arne-heil 

Currently, Elasticsearch is an optional dependency for the API Gateway (you could rely on other reporters like TCP, file, ...).
But it's still not at the Management API level.

Why disabling ES would need a minimal footprint (for the Management API) ?

**Comment by arne-heil** [comment]
> > Please implement this feature. We could really use a minimal footprint with at least Elasticsearch as an optional choice. Thanks in advance smile
> 
> Hi @arne-heil
> 
> Currently, Elasticsearch is an optional dependency for the API Gateway (you could rely on other reporters like TCP, file, ...).
> But it's still not at the Management API level.
> 
> Why disabling ES would need a minimal footprint (for the Management API) ?

We would like to enable and disable ES when needed. Otherwise we have to maintain a ES instance only for Gravitee usage and/or disable the Management API after setting up and deploying our API.

**Comment by brasseld** [comment]
> We would like to enable and disable ES when needed. Otherwise we have to maintain a ES instance only for Gravitee usage and/or disable the Management API after setting up and deploying our API.

Meaning that you're happy not having any kind of analytics from Gravitee, right?

**Comment by arne-heil** [comment]
> > We would like to enable and disable ES when needed. Otherwise we have to maintain a ES instance only for Gravitee usage and/or disable the Management API after setting up and deploying our API.
> 
> Meaning that you're happy not having any kind of analytics from Gravitee, right?

In general, yes. We don't have any complains about Gravitee in combination with our API  and analytics would be a nice optional feature for us.

**Comment by ydbondt** [comment]
If there is any intention to apply this, you can reopen the pull request.

**Pull request closed without merging**